include/mata/nfa/delta.hh

```diff
@@ -134,6 +152,7 @@ public:
 
     void clear() { posts.clear(); }
 
+    //Still fishy, what do we mean by size here? Probably should be just the size of the vector. However, below we call it num_of_states.
     void increase_size(size_t n) {
```

Comment:
@kilohsakul What do you think about the proposal for `Delta::allocate(size_t num_of_states)`?

Reply:
Keeping `reserve`, removing `increase_size()`. Since `Delta` returns an empty state post for states not appearing in transitions, `increase_size()` does not have any meaningful uses. Unless we explicitly initialize a state post with some transition, non-existent state post and empty state post created with `increase_size()` return the same thing and behave exactly the same.

Reply:
Scratch the previous message. If we intend to have `add_state()`, we have to reflect the call to `add_state()` in the number of states returned by `num_of_states()`. Hence, we cannot only reserve the states, we need to truly allocate empty state posts.

Keeping `reserve()`. `increase_size()` renamed to `allocate(size_t num_of_states)`. This should be clear, and adding states works as currently intended. Whether adding states manually is a good idea remains to be seen.